cmd/kata-pkgsync/cli/main.go

```diff
@@ -0,0 +1,231 @@
+	if len(flag.Args()) > 0 {
+		configFile = flag.Args()[0]
+	} else {
+		configFile = defaultConfig
+	}
+	logrus.Debugf("Using config file %s", configFile)
+
+	cfg, err := getConfig(configFile)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to read config")
+		os.Exit(-1)
+	}
+	logrus.Debugf("config: %#v", cfg)
+
+	pcRepo := path.Join(cfg.Packagecloud.Auth.User, cfg.Packagecloud.Repo)
+	logrus.WithField("Packagecloud repo", pcRepo).Info("Retrieving list of available files")
+
+	var pc PCClient
+	if err := pc.PackagecloudClient(cfg.Packagecloud.Auth.User, cfg.Packagecloud.Auth.Token, cfg.Packagecloud.Repo); err != nil {
+		logrus.WithError(err).Error("Failed to create a Packagecloud client instance")
+		os.Exit(-1)
+	}
+
+	pcPackages, err := pc.PackagecloudList()
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"Packagecloud repo": pcRepo,
+			"error":             err,
+		}).Error("Failed to retrieve available files")
+		os.Exit(-1)
+	}
+
+	// lookup table for pcPackages packages that should NOT be deleted from Packagecloud
+	pcKeepPackagesLUT := make([]bool, len(pcPackages))
+
+	var projects []obsgo.Project
+	for n, p := range cfg.OBSProjects {
+		proj := obsgo.Project{
+			User:     p.Auth.User,
+			Password: p.Auth.Password,
+		}
+
+		// Kata projects names format is "project:release:architecture"
+		for _, arch := range p.Archs {
+			for _, release := range p.Releases {
+				proj.Name = strings.Join([]string{n, arch, release}, ":")
+				projects = append(projects, proj)
+			}
+		}
+
+		// Fall-back non-Kata project name format
+		if len(projects) == 0 {
+			proj.Name = n
+			projects = append(projects, proj)
+		}
+	}
+
+	for _, proj := range projects {
+		logrus.WithField("OBS project", proj.Name).Info("Retrieving packages info")
+
+		obsPackages, err := proj.FindAllPackages()
+		if err != nil {
+			logrus.WithError(err).WithField("project", proj.Name).Error("Failed to get OBS packages")
+			os.Exit(-1)
+		}
+
+		logrus.WithField("OBS project", proj.Name).Infof("Found %d packages", len(obsPackages))
+
+		totalXferred := 0
+		for _, obsPkg := range obsPackages {
+			pcDistro, found := cfg.DistroMapping[obsPkg.Repo]
+			if !found {
+				logrus.WithFields(logrus.Fields{
+					"OBS Repo": obsPkg.Repo,
+				}).Warnf("No mapped Packagecloud distro specified")
+				os.Exit(-1)
+			}
+
+			if pcDistro == "" {
+				logrus.WithFields(logrus.Fields{
+					"package":  obsPkg.Name,
+					"OBS Repo": obsPkg.Repo,
+				}).Warnf("Repo not supported by Packagecloud")
+				continue
+			}
+
+			var xferBins []obsgo.PkgBinary
+			for _, src := range obsPkg.Files {
+				found = false
+				for i, dst := range pcPackages {
+					logrus.Debugf("Checking pkg %s %s VS %s %s", src.Filename, pcDistro, dst.Filename, dst.DistroVersion)
+					if pcDistro == dst.DistroVersion && src.Filename == dst.Filename {
+						logrus.Debugf("%s already on Packagecloud %s/%s", src.Filename, pcRepo, pcDistro)
+						found = true
+						pcKeepPackagesLUT[i] = true
```

Comment:
This is very deeply indented code. Could you maybe break `main()` up into smaller functions? That would keep the static linters happy too I suspect ;)